caixa/caixa_test.go

```diff
@@ -308,3 +308,55 @@ func TestCaixaValidateInterestWithNilFees(t *testing.T) {
 
 	assert.Nil(t, caixaValidateInterest(input))
 }
+
+func TestTemplateRequestCaixa_WhenRequestWithSacadorAvalistaIsCNPJ_ParseSuccessful(t *testing.T) {
+	f := flow.NewFlow()
+	input := newStubBoletoRequestCaixa().WithRecipientName("Recipient Test Name").WithRecipientDocumentType("CNPJ").Build()
+
+	b := fmt.Sprintf("%v", f.From("message://?source=inline", input, getRequestCaixa(), tmpl.GetFuncMaps()).GetBody())
+	no_content := test.GetNode(b, "SACADOR_AVALISTA")
+
+	assert.Contains(t, no_content, "<RAZAO_SOCIAL>Recipient Test Name</RAZAO_SOCIAL>", "Erro no mapeamento do campo Recipient")
+}
+
+func TestTemplateRequestCaixa_WhenRequestWithSacadorAvalistaIsCPF_ParseSuccessful(t *testing.T) {
+	f := flow.NewFlow()
+	input := newStubBoletoRequestCaixa().WithRecipientName("Recipient Test Name").WithRecipientDocumentType("CPF").Build()
+
+	b := fmt.Sprintf("%v", f.From("message://?source=inline", input, getRequestCaixa(), tmpl.GetFuncMaps()).GetBody())
+	node_content := test.GetNode(b, "SACADOR_AVALISTA")
+
+	assert.Contains(t, node_content, "<NOME>Recipient Test Name</NOME>", "Erro no mapeamento do campo Recipient")
+}
+
+func TestTemplateRequestCaixa_WhenRequestWithSacadorAvalistaIsCPF_ParseFailed(t *testing.T) {
+	f := flow.NewFlow()
+	input := newStubBoletoRequestCaixa().WithRecipientName("Recipient Test Name").WithRecipientDocumentType("CNPJ").Build()
+
+	b := fmt.Sprintf("%v", f.From("message://?source=inline", input, getRequestCaixa(), tmpl.GetFuncMaps()).GetBody())
+	node_content := test.GetNode(b, "SACADOR_AVALISTA")
+
+	assert.NotContains(t, node_content, "<NOME>Recipient Test Name</NOME>", "Erro no mapeamento do campo Recipient")
+}
+
+func TestTemplateRequestCaixa_WhenRequestWithSacadorAvalistaIsCNPJ_ParseFailed(t *testing.T) {
+	f := flow.NewFlow()
+	input := newStubBoletoRequestCaixa().WithRecipientName("Recipient Test Name").WithRecipientDocumentType("CPF").Build()
+
+	b := fmt.Sprintf("%v", f.From("message://?source=inline", input, getRequestCaixa(), tmpl.GetFuncMaps()).GetBody())
+	node_content := test.GetNode(b, "SACADOR_AVALISTA")
+
+	assert.NotContains(t, node_content, "<RAZAO_SOCIAL>Recipient Test Name</RAZAO_SOCIAL>", "Erro no mapeamento do campo Recipient")
+}
+
+func TestTemplateRequestCaixa_WhenRequestWithSacadorAvalistaDocumentInNotValidCPF_ParseFailed(t *testing.T) {
+	doc := "00732159000109"
```

Comment:
uma sugestão pra deixar mais claro o nome dessa variável 😀 
```suggestion
	cnpjDocument:= "00732159000109"
```